src/DynamoRevit/DynamoRevit.cs

```diff
@@ -149,6 +154,22 @@ public Result Execute(ExternalCommandData commandData, ref string message, Eleme
             return Result.Succeeded;
         }
 
+        private static void OnApplicationIdle(object sender, IdlingEventArgs e)
+        {
+            if (!idleActions.Any())
+                    return;
```

Comment:
@sharadkjaiswal, I think `Any()` will result in executing the enumerator, so if anyone inserts a new item, it may invalidate the enumerator and cause exception to be thrown. We should use simple property `idleActions.Count` instead.